### PR TITLE
Setup fastlane automation

### LINKED ISSUE #318: Fastlane Automation
https://fastlane.tools/
* GitHub Actions for CI integration
* Meta data for Google Play and App Store
* Meta data for app permissions, e.g. `NSLocationWhenInUseUsageDescription`
* Screenshot automation
* App Store and Google Play binary upload
* Localization support

### PR DESCRIPTION
**Pull request by augustinreille**
**Please follow this checklist. Please check each appropriate box (put an 'x' or check it after creating the PR).**
- [x] Provided detailed pull request description and a succinct title (consider template below for guidance).
- [ ] Tested your changes, especially after any code review iterations.
- [ ] Included any relevant screenshots of UI updates.
- [x] Followed the [Contributor Guidelines](https://github.com/WorldHealthOrganization/app/blob/master/docs/CONTRIBUTING.md).
- [ ] Verified all contributions are properly licensed pursuant to the [LICENSE](https://github.com/WorldHealthOrganization/app/blob/master/LICENSE) file in the root of the repository.
- [x] Verified your name is in the [content/credits.yaml](https://github.com/WorldHealthOrganization/app/blob/master/content/credits.yaml) file (if you want it to be).

After all boxes above are checked, request and receive an Approved review from any team member knowledgable in the area (TODO team member list).  Once approved, the team member will assign your review to a Committer or use the `needs-merge` label.

## What does this PR accomplish?

This PR adds [fastlane](https://fastlane.tools/) support. It allows to automatically deploy app on both stores (Apple Store and Play Store). Closes #318.

## TODOS

- [x] initialize Fastlane
- [x] use correct package name
- [x] use correct bundle ID
- [ ] integrate Google Console JSON file (see [this link](https://docs.fastlane.tools/actions/supply/#setup))
- [ ] use correct `apple_id` (Apple email address)
- [ ] use correct `itc_team_id` (App Store Connect Team ID)
- [ ] use correct `team_id` (Developer Portal Team ID)
- [ ] setup Github Actions to build project for iOS
- [ ] setup Github Actions to build project for Android
- [ ] setup Github Actions to run fastlane

## Usage 

These are the scripts which are to be integrated to CI in order to deploy apps using fastlane :

#### Internal test

```
    - flutter build apk --split-per-abi
    - cd android/
    - gem install bundler
    - bundle install
    - bundle exec fastlane upload_to_play_store_internal
```

#### From internal test to alpha

```
    - cd android/
    - gem install bundler
    - bundle install
    - bundle exec fastlane promote_internal_to_alpha
```

#### From alpha to production

```
    - cd android/
    - gem install bundler
    - bundle install
    - bundle exec fastlane promote_alpha_to_production
```

#### All at once
```
    - flutter build apk --split-per-abi
    - cd android/
    - gem install bundler
    - bundle install
    - bundle exec fastlane upload_to_play_store_internal
    - bundle exec fastlane promote_internal_to_alpha
    - bundle exec fastlane promote_alpha_to_production
```

#### TestFlight

```
    - flutter build ios
    - cd ios/
    - gem install bundler
    - bundle install
    - bundle exec fastlane beta
```

#### From TestFlight to production

```
    - cd ios/
    - gem install bundler
    - bundle install
    - bundle exec fastlane submit_review
```

#### All at once

```
    - flutter build ios
    - cd ios/
    - gem install bundler
    - bundle install
    - bundle exec fastlane beta
    - bundle exec fastlane submit_review
```

### CODE REVIEW

File: client/flutter/android/fastlane/Appfile [hunk 1]
```diff
@@ -0,0 +1 @@
+package_name("org.who.WHOMyHealth") # e.g. com.krausefx.app
```
Comment (SamMousa) [inline client/flutter/android/fastlane/Appfile]:
```suggestion
package_name("org.who.WHOMyHealth")
```

File: client/flutter/android/fastlane/Fastfile [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      skip_upload_changelogs: true,
+      skip_upload_metadata: true,
+      skip_upload_images: true,
+      skip_upload_screenshots: true,
+      apk_paths: [
+        '../build/app/outputs/apk/release/app-armeabi-v7a-release.apk',
+        '../build/app/outputs/apk/release/app-arm64-v8a-release.apk',
+        '../build/app/outputs/apk/release/app-x86_64-release.apk'
+      ],
+      json_key_data: ${{ secrets.JSON_KEY_DATA }}
```
Comment (SamMousa) [inline client/flutter/android/fastlane/Fastfile]:
```suggestion
      json_key_data: ${{ secrets.GOOGLE_STORE_KEY }}
```

File: client/flutter/android/fastlane/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+fastlane documentation
+================
+# Installation
+
+Make sure you have the latest version of the Xcode command line tools installed:
+
+```
+xcode-select --install
```
Comment (SamMousa) [inline client/flutter/android/fastlane/README.md]: Do we need xcode for android?

File: client/flutter/android/fastlane/report.xml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (SamMousa) [inline client/flutter/android/fastlane/report.xml]: I'm not sure this file needs to be in the repo.

File: client/flutter/ios/fastlane/Appfile [hunk 5]
```diff
@@ -0,0 +1,8 @@
+app_identifier("int.who.WHOMyHealth") # The bundle identifier of your app
```
Comment (SamMousa) [inline client/flutter/ios/fastlane/Appfile]:
```suggestion
app_identifier("int.who.WHOMyHealth")
```

**Comment by SamMousa** [comment]
For all variables used by fastlane, I think you should use environment variables.
We can then populate them with secrets.

File: client/flutter/android/fastlane/README.md (hunk 3 above)
Comment (areille) [reply client/flutter/android/fastlane/README.md]: No, you're right. This is a generated file. This is definitly an issue with fastlane. These README files could be removed.

File: client/flutter/android/fastlane/report.xml (hunk 4 above)
Comment (areille) [reply client/flutter/android/fastlane/report.xml]: This is also a generated file. Will remove it

File: client/flutter/ios/fastlane/Fastfile [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  desc "Push a new beta build to TestFlight"
+  lane :beta do
+    build_app(workspace: "Runner.xcworkspace", scheme: "Runner")
+    upload_to_testflight
+  end
+
+  desc "Promote the beta build to App Store"
+  lane :submit_review do
+    deliver(
+      submit_for_review: true,
```
Comment (KrauseFx) [inline client/flutter/ios/fastlane/Fastfile:28]: Realistically on the first release, setting `submit_for_review` to `true` will likely fail, as AppStore Connect has follow-up questions around encryption and ad tracking. I'd recommend doing at least one release manually before automating the actually `submit_for_review`

Comment (augustinreille) [reply client/flutter/ios/fastlane/Fastfile:28]: Totally agree.

File: client/flutter/ios/fastlane/Appfile [hunk 7]
```diff
@@ -0,0 +1,8 @@
+app_identifier("int.who.WHOMyHealth")
+apple_id("") # Your Apple email address
+
+itc_team_id("") # App Store Connect Team ID
+team_id("") # Developer Portal Team ID
```
Comment (brunobowden) [inline client/flutter/ios/fastlane/Appfile:5]: @KrauseFx @joshdholtz - is there any issue with having WHO's `itc_team_id` and `team_id` in a public repo? I see a number of other team ids listed elsewhere in open source code

File: client/flutter/ios/fastlane/Fastfile [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+      automatic_release: true,
```
Comment (brunobowden) [inline client/flutter/ios/fastlane/Fastfile:29]: We'd like to do automatic upload but manual confirmation of release. Will this automatically release a new version based once approved? If so, then please change this to `false`. Given lots of localized binaries, we'd probably need a separate command that did the actual release. I'm not sure how many of the binaries will go through Apple review.

File: client/flutter/ios/fastlane/Appfile (hunk 7 above)
Comment (augustinreille) [reply client/flutter/ios/fastlane/Appfile:5]: We could use Github Actions secret variables for these.

File: client/flutter/ios/fastlane/Fastfile (hunk 8 above)
Comment (augustinreille) [reply client/flutter/ios/fastlane/Fastfile]: Yes it would automatically release version once approved. Change done.

**Comment by augustinreille** [comment]
@brunobowden Should we already setup fastlane to automatically update screenshots to the stores ?

**Comment by SamMousa** [comment]
At the very least we can make screenshots automated, even if not yet used in the store. That way we can publish them to Web and use them in docs.

File: client/flutter/android/fastlane/Appfile [hunk 9]
```diff
@@ -0,0 +1 @@
+package_name("org.who.WHOMyHealth")
```
[the remark this answers is not in the document]
Comment (augustinreille) [reply client/flutter/android/fastlane/Appfile:1]: Should `Appfile` be deleted ?

[the remark this answers is not in the document]
Comment (augustinreille) [reply client/flutter/android/fastlane/Appfile:1]: Should I do the same for ios ?

**Comment by augustinreille** [comment]
@joshdholtz thanks for these suggestions 😃

File: client/flutter/android/fastlane/Appfile (hunk 9 above)
[the remark this answers is not in the document]
Comment (joshdholtz) [reply client/flutter/android/fastlane/Appfile:1]: Yup yup, you can delete the `Appfile` 😊 And the same can probably be done for `iOS` for now as well (at least for now unless we need it for a workaround with GitHub secrets 🙃)

File: client/flutter/android/Gemfile [hunk 10]
```diff
@@ -0,0 +1,3 @@
+source "https://rubygems.org"
+
+gem "fastlane"
```
Comment (epicfaace) [inline client/flutter/android/Gemfile:3]:
```suggestion
gem "fastlane"

```

File: client/flutter/android/fastlane/.env [hunk 11]
```diff
@@ -0,0 +1,5 @@
+SUPPLY_PACKAGE_NAME="org.who.WHOMyHealth"
+SUPPLY_SKIP_UPLOAD_CHANGELOGS=true
+SUPPLY_SKIP_UPLOAD_METADATA=true
+SUPPLY_SKIP_UPLOAD_IMAGES=true
+SUPPLY_SKIP_UPLOAD_SCREENSHOTS=true
```
Comment (epicfaace) [inline client/flutter/android/fastlane/.env:5]:
```suggestion
SUPPLY_SKIP_UPLOAD_SCREENSHOTS=true

```

File: client/flutter/ios/fastlane/Fastfile [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+platform :ios do
+  desc "Push a new beta build to TestFlight"
+  lane :beta do
+    build_app(workspace: "Runner.xcworkspace", scheme: "Runner")
+    upload_to_testflight
+  end
+
+  desc "Promote the beta build to App Store"
+  lane :submit_review do
+    deliver()
```
Comment (joshdholtz) [inline client/flutter/ios/fastlane/Fastfile:27]:
```suggestion
    deliver
```

Nit pick but don’t need parenthesis anymore here 😇

Comment (augustinreille) [reply client/flutter/ios/fastlane/Fastfile:27]: I was wondering about this 😉

**Comment by hspinks** [comment]
> I will need some help to setup screenshots automation. @SamMousa @brunobowden could this be done in another PR ?

@augustinreille yes, a separate PR for that change is fine by me 👍

**Comment by epicfaace** [comment]
@brunobowden I believe we need your approval to go forward with this (as you requested changes)

**Comment by brunobowden** [comment]
@crazybob - PTAL, we should get this in ASAP

**Pull request merged**